include/flatbuffers/reflection_generated.h

```diff
@@ -159,6 +166,12 @@ inline flatbuffers::Offset<Type> CreateType(
 }
 
 struct KeyValue FLATBUFFERS_FINAL_CLASS : private flatbuffers::Table {
+  flatbuffers::Table* AsTable() {
```

Comment:
I don't think I see the point of this.. this adds a lot of code, and all it does is cast to the inaccessible base class? I understand the current `IsFieldPresent` can cast anything, which is less safe, but there has to be a better solution. Maybe we can do another inheritance level for example, make a `BaseTable` that inherits from `private flatbuffers::Table` and then generated classes can publicly inherit from `BaseTable` ? Once we have that, there may be other places in the code base that currently use `Table` that can then use `BaseTable` instead.

Reply:
That seems reasonable although I'm unsure about the naming.  The hierarchy will then be `MyTable -> public BaseTable -> private Table`.  Would it be okay to rename `Table` to `BaseTable` or something like `InnerTable` so we'll get `MyTable -> public Table -> private BaseTable`?

Reply:
I'd be hesitant to rename `Table`, as that will break a fair bit of code. Maybe call the middle level `GeneratedTable`, or is that too verbose? Whatever makes sense.